ast/check.go

```diff
@@ -652,72 +657,50 @@ func (rc *refChecker) checkRef(curr *TypeEnv, node *typeTreeNode, ref Ref, idx i
 
 	head := ref[idx]
 
```

Comment:
These are the type checker changes 👇 It's also the piece I'm most uncertain about 😅